apps/metadata-editor/src/app/records/my-records/my-records.component.ts

```diff
@@ -11,8 +14,29 @@ import { SearchFacade } from '@geonetwork-ui/feature/search'
   standalone: true,
   imports: [CommonModule, TranslateModule, RecordsListComponent],
 })
-export class MyRecordsComponent {
-  constructor(public searchFacade: SearchFacade) {
+export class MyRecordsComponent implements OnInit, OnDestroy {
+  private sub: Subscription
+  private ownerId: string
+
+  constructor(
+    public searchFacade: SearchFacade,
+    private authService: AuthService,
+    private router: EditorRouterService
+  ) {}
+
+  ngOnInit() {
     this.searchFacade.resetSearch()
+    this.sub = this.authService.user$.subscribe((user) => {
+      this.ownerId = user.id
+      this.searchFacade.updateFilters({ owner: user.id })
+    })
+  }
+
+  getDatahubUrl(): string {
+    return `${this.router.getDatahubSearchRoute()}?owner=${this.ownerId}`
```

Comment:
This assumes that the datahub URL doesn't contain any query param. It's generally safer to use the `URL` class and its API to modify urls